src/sisl/io/siesta/vibra.py

```diff
@@ -0,0 +1,257 @@
+        #  _state is:
+        #        -1 : the file-descriptor has just been opened (i.e. in front of header)
+        #         0 : it means that the file-descriptor is in front of k point information
+        self._state = -1
+        self._ik = 0
+
+    def _setup_parsing(self, close=True):
+        """Gets all the things needed to parse the wfsx file.
+
+        Parameters
+        -----------
+        close: bool, optional
+            Whether the file unit should be closed afterwards.
+        """
+        self._open()
+        # Read the sizes relevant to the file.
+        # We also read whether there's only gamma point information or there are multiple points
+        if self._state == -1:
+            self._sizes = self._r_sizes()
+            self._state = 0
+
+        if close:
+            self._close()
+
+    def _r_sizes(self):
+        """ Determine the dimension of the data stored in the vectors file
+        """
+        pos = self.fh.tell()
+        # Skip past header
+        self.readline() # Empty line
+        self.readline() # K point
+        self.readline() # Eigenmode index
+        self.readline() # Frequency
+        self.readline() # Eigenmode header
+        
+        # Determine number of atoms (i.e. rows per mode)
+        natoms = 0
+        while True:
+            line = self.readline()
+            if line.startswith('Eigenmode'):
+                break
+            else:
+                natoms += 1
+        self._natoms = natoms
+
+        # Determine number of modes
+        nmodes = 1
+        while True:
+            line = self.readline()
+            if line.startswith('k            ='):
+                break 
+            elif line.startswith('Eigenvector'):
+                nmodes += 1
+        self._nmodes = nmodes
+
+        # Rewind
+        self.seek(pos)
+
+    def _r_next_eigenmode(self):
+        """Reads the next phonon eigenmode in the vectors file.
+
+        Parameters
+        ----------
+        ik: integer
+            the (python) k index of the next eigenstate.
+
+        Returns
+        --------
+        EigenmodePhonon:
+            The next eigenmode.
+        """
+        # Skip empty line at the head of the file
+        self.readline()
+            
+        k = list(map(float, self.readline().split()[2:]))
+        if len(k) == 0:
+            raise GeneratorExit
+        # Read first eigenvector index 
+        
+        # Determine number of atoms (i.e. rows per mode)
+        state = np.empty((self._nmodes, self._natoms, 3), dtype=np.complex128)
+        c = np.empty(self._nmodes, dtype=np.float64)
+        for imode in range(self._nmodes):
+            line = self.readline()
+            if line.strip() == '':
+                # End of block for current k is indicated by empty line
+                break
+            
+            # Read frequency
+            c[imode] = float(self.readline().split('=')[1].strip()) * _hc_eV
```

Comment:
do the `_hc_eV` conversion in the end, faster.

Also `strip` shouldn't be needed